File: tools/research-probes/2026-09-13/read_connector_routing_cdp.mjs
const targets = await (await fetch('http://127.0.0.1:9230/json/list')).json();
const t = targets.find(x => x.url === 'chrome-extension://fnjbijbhcehgoglobkicibfpcmddlggg/service-worker-loader.js');
if (!t) throw new Error('Browser Connector service worker target not found');
const ws = new WebSocket(t.webSocketDebuggerUrl);
const done = new Promise((resolve, reject) => {
  ws.addEventListener('open', () => ws.send(JSON.stringify({id:1,method:'Runtime.evaluate',params:{expression:"chrome.storage.local.get('catalystRoutingDebug')",awaitPromise:true,returnByValue:true}})));
  ws.addEventListener('message', ev => { const msg=JSON.parse(ev.data); if (msg.id===1) { console.log(JSON.stringify(msg,null,2)); resolve(); } });
  ws.addEventListener('error', reject);
});
await done;
ws.close();

File: tools/research-probes/2026-09-13/read_connector_routing_cdp.py
import json, urllib.request
import websocket

targets=json.load(urllib.request.urlopen('http://127.0.0.1:9230/json/list'))
t=next(x for x in targets if x.get('url')=='chrome-extension://fnjbijbhcehgoglobkicibfpcmddlggg/service-worker-loader.js')
ws=websocket.create_connection(t['webSocketDebuggerUrl'], timeout=5, origin='http://127.0.0.1:9230')
ws.send(json.dumps({'id':1,'method':'Runtime.evaluate','params':{'expression':"chrome.storage.local.get('catalystRoutingDebug')",'awaitPromise':True,'returnByValue':True}}))
while True:
    msg=json.loads(ws.recv())
    if msg.get('id')==1:
        print(json.dumps(msg, indent=2))
        break
ws.close()
